endpoint/endpoint.go

```diff
@@ -116,7 +117,11 @@ func (s *Sender) ListFilesystemVersions(ctx context.Context, r *pdu.ListFilesyst
 	if err != nil {
 		return nil, err
 	}
-	fsvs, err := zfs.ZFSListFilesystemVersions(ctx, lp, zfs.ListFilesystemVersionsOptions{})
+
+	fsvs, err := zfs.ZFSListFilesystemVersions(ctx, lp, zfs.ListFilesystemVersionsOptions{
+		ShortnamePrefix: s.config.JobSnapPrefix,
```

Comment:
Maybe we should have something more flexible than just prefix matching?
See my ideas in https://github.com/zrepl/zrepl/issues/403#issuecomment-820698946

Reply:
Sure there is space for improvements, I’ll think about it but I can’t promise anything in a short time.